### ISSUE TITLE
Man page cosmetics

### ISSUE THREAD

**Issue by maxried** [fish-shell/fish-shell#4016]
```
~$ fish --version                                         
fish, version 2.5.0-368-g8814f34d-dirty
```
```
~$ uname -a                                                                                                                                                  
Darwin nathan 16.5.0 Darwin Kernel Version 16.5.0: Fri Mar  3 16:52:33 PST 2017; root:xnu-3789.51.2~3/RELEASE_X86_64 x86_64
```

fish's man pages always seem to have the command name written twice in the first paragraph. Also, the following line is nothing but a dash.

```
jobs(1)                              fish                              jobs(1)

NAME
       jobsjobs - print currently running jobs
        -

   Synopsis
       jobs [OPTIONS] [PID]
```

or 

```
fish(1)                              fish                              fish(1)

NAME
       fishfish - the friendly interactive shell
        -

   Synopsis
       fish [OPTIONS] [-c command] [FILE [ARGUMENTS...]]
```

On the other hand system provided man pages don't have this:

```
IFCONFIG(8)               BSD System Manager's Manual              IFCONFIG(8)

NAME
     ifconfig -- configure network interface parameters

SYNOPSIS
     ifconfig [-L] [-m] [-r] interface [create] [address_family] [address
              [dest_address]] [parameters]
```

and 

```
YES(1)                    BSD General Commands Manual                   YES(1)

NAME
     yes -- be repetitively affirmative

SYNOPSIS
     yes [expletive]
```

**Comment by zanchey** [comment]
I'm pretty sure this is due to changes in Doxygen - I see this output with 1.8.13 on my OS X system, but not from 1.8.8 on my Debian system.

**Comment by krader1961** [comment]
FWIW, I have Doxygen 1.8.10, 1.8.11, 1.8.12, and 1.8.13 on my macOS systems thanks to Homebrew keeping old versions installed by default. All of them produce the wrong output. I also have doxygen 1.8.6 on Ubuntu 14.04 and it too produces the wrong output. Not sure what to make of that given that @zanchey says doxygen 1.8.8 produces the correct output. It's possible that the current behavior is the correct behavior and it was broken in the 1.8.8 release and subsequently fixed.

**Comment by krader1961** [comment]
I spent over an hour trying to figure out a solution by googling and trying semi-random changes. Note that our man pages are formatted in other odd ways. Compare `man fish` to `man bash`. Hint: Notice that sections like "SYNOPSIS" are top-level sections in other man pages and subsections in fish man pages.

We don't really need Doxygen since we are not creating API documentation from source code comments. I feel quite strongly that we would be better served by practically any alternative. However, "any" does not include Texinfo since it does not directly support man pages. Which makes me sad since I love TeX (and LaTeX) for writing beautiful documents.  There are tools, like [LaTeX2man](http://mirrors.ibiblio.org/CTAN/support/latex2man/latex2man.html) which would allow us to use LaTeX as our documentation markup format. There is also [Pandoc](http://pandoc.org/) which is a swiss army knife for converting between different markup languages.

Another option is to use Markdown as our markup format. Since most contributors to the project are familiar with it by virtue of using Github to host the project this is likely a good choice. Markdown, however, has some deficiencies. Notably it doesn't support semantic markup. It's a lot like HTML in that regard. However, it's not obvious that is a fatal flaw for our needs.

While I would be happy to see this specific issue fixed by someone I'm not going to spend another second on it. Because I think improving our documentation is better served by switching from Doxygen to another mechanism.

**Comment by geoffmacintosh** [comment]
I know a bit about LaTeX (although not much) and I'd be happy to help convert documentation if that would be useful.

**Comment by zanchey** [comment]
I looked at some alternatives in #980.

**Comment by krader1961** [comment]
Thanks, @zanchey, for pointing out that our dependency on Doxygen has been a source of pain for several years. The fact that these types of issues with Doxygen keep recurring, and there have been several more I can recall beyond those referenced in this issue, is a strong argument for switching to an alternative.

**Comment by faho** [comment]
Fixed by 8e93041205b915eeb285c49220d942edf820ba38.
